### PR TITLE
Add method reset password his functionality

### PR DESCRIPTION
**Pull request by valetobares11**

### CODE REVIEW

**Comment by saiema** [comment]
Is this a PR for adding the password reset functionality?

File: src/main/java/unrc/dose/User.java [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
-*
-* Table name: users
-*
-*  id              :integer(11)    not null, primary key
-*  username        :varchar(20)    not null,
-*  password        :varchar(20)    not null,
-*  email_address   :varchar(50)    not null,
-*  admin           BOOLEAN not null default 0
-*
-**/
```
Comment (saiema) [inline src/main/java/unrc/dose/User.java]: Why did you remove these comments? These should be present to describe the associated table

File: src/main/java/unrc/dose/User.java [hunk 2]
```diff
@@ -1,19 +1,39 @@
...
+
 
 package unrc.dose;
 
 import org.javalite.activejdbc.Model;
+import org.javalite.activejdbc.LazyList;
 
 public class User extends Model {
+	static final String  PSW_RESET = "password reset";
+	static final String  PSW_NOT_RESET = "the email is not registered in the DB";
```
Comment (saiema) [inline src/main/java/unrc/dose/User.java]: These attributes should be private, unless they are used outside this class, but since there is no documentation attached I can't know if they will.

File: src/main/java/unrc/dose/User.java [hunk 3]
```diff
@@ -1,19 +1,39 @@
...
+		LazyList<User> users = User.where("email_address = ?", email);
+		User user = new User();
+		if(users.size() > 0) {
+			user = users.get(0);
+			user.set("password",pass);
+			user.saveIt();
+			return PSW_RESET;
+		}
+		return PSW_NOT_RESET;
+	}
```
Comment (saiema) [inline src/main/java/unrc/dose/User.java]: There is no documentation attached to this method, you should always document at least all public attributes and methods. Taken into account that after a comma there should be a blank space, and between an `if` at its condition, there should also be a blank space.

The new instance in line 27, that is, `User user = new User();`, is never used, likewise, the `user` variable is only used in the `then` part of the if statement, and as such this variable should be local to this if statement.

Finally, consider that the returning value should be a `true` \ `false` value.

File: src/test/java/unrc/dose/UserTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    	Spark.stop();
+    	Base.close();
+  	}
+
+  	/**
+	 * test that verifies that the password has been modified successfully
+	 * @result the email is not registered in the DB
+	*/
+	@Test
+	public void resetPasswordNotSuccessfully(){
```
Comment (saiema) [inline src/test/java/unrc/dose/UserTest.java]: The name should be `updatePasswordUnsuccessful`, but also you should consider that there are several possible reasons for a password update to fail.
 - User doesn't exist (no user with provided email)
 - The provided password is the same as the current password
 - The provided password is empty or doesn't comply with password policies (e.g.: minimum length)

File: src/test/java/unrc/dose/UserTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		assertEquals("the email is not registered in the DB" ,User.updatePassword(email,pass));
+	}
+
+
+	/**
+	 * test that verifies that the password could not be modified
+	 * @result password reset
+	*/
+	@Test
+	public void resetPasswordSuccessfully(){
```
Comment (saiema) [inline src/test/java/unrc/dose/UserTest.java]: The name should be `updatePasswordSuccessful`

File: src/test/java/unrc/dose/UserTest.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		String email = "JohnConnor@gmail.com";
+		String pass = "VeryHardPass";
+
+		assertEquals("password reset",User.updatePassword(email,pass));
```
Comment (saiema) [inline src/test/java/unrc/dose/UserTest.java]: Remember that after a comma there should be a blank space

File: src/test/java/unrc/dose/UserTest.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  	/**
+	 * test that verifies that the password has been modified successfully
+	 * @result the email is not registered in the DB
+	*/
+	@Test
+	public void resetPasswordNotSuccessfully(){
+		String email = "juanPerez@gmail.com";
+		String pass = "VeryHardPass";
+
+		assertEquals("the email is not registered in the DB" ,User.updatePassword(email,pass));
```
Comment (saiema) [inline src/test/java/unrc/dose/UserTest.java]: Remember that after a comma there should be a blank space

File: src/test/java/unrc/dose/UserTest.java [hunk 8]
```diff
@@ -108,6 +119,71 @@ public void deleteUserUnsuccessful(){
...
 		assertEquals(false, User.disableUser(username, password));
 	}
+	/**
+	 * test that verifies that the password has been modified successfully
+	 * @result false bescause the email is not registered in the DB
+	 */
+	@Test
+	public void updatePasswordUnSuccessfully() {
+		String email = "juanPerez@gmail.com";
+		String pass = "NewPass";
```
Comment (pponzio) [inline src/test/java/unrc/dose/UserTest.java]: I do not like this style very much. I think constants should be used directly as parameters in this case.
